descartes_light/solvers/include/descartes_light/solvers/ompl/ompl_solver.h

```diff
@@ -0,0 +1,72 @@
+#ifndef DESCARTES_LIGHT_SOLVERS_OMPL_OMPL_SOLVER_H
+#define DESCARTES_LIGHT_SOLVERS_OMPL_OMPL_SOLVER_H
+
+#include <descartes_light/solvers/bgl/bgl_solver.h>
+#include <descartes_light/solvers/ompl/descartes_space.h>
+#include <ompl/base/Planner.h>
+#include <ompl/geometric/SimpleSetup.h>
+
+namespace descartes_light
+{
+/**
+ * @brief BGL solver implementation that constructs vertices and edges in the build function and uses Dijkstra's
+ * algorithm with a default visitor to search the graph
+ */
+template <typename FloatType>
+class BGLOMPLSolver : public BGLSolverBaseSVDE<FloatType>
+{
+public:
+  using BGLSolverBaseSVDE<FloatType>::BGLSolverBaseSVDE;
+  BGLOMPLSolver(double max_dist,
+                double planning_time,
+                unsigned num_threads = std::thread::hardware_concurrency(),
+                double rung_to_rung_dist = 100000)
+    : BGLSolverBaseSVDE<FloatType>(num_threads)
+    , max_dist_(max_dist)
+    , planning_time_(planning_time)
+    , rung_to_rung_dist_(rung_to_rung_dist)
+  {
+  }
+
+  SearchResult<FloatType> ompl_search(std::shared_ptr<ompl::base::Planner> ompl_planner);
+
+  void initOMPL();
```

Comment:
Do these need to be separate from the constructor and/or `search` method?

Reply:
I could combine `initOMPL()` with a redefinition of the constructor.

The `ompl_search` method is setup this way because I wanted the classes that inherit from it to easily call it without having to re-write everything in it. Also, it requires an input of a planner type, so it can't just inherit from `search`. This can't be just templated on `ompl::base::Planner` either because each planner has a unique set of parameters, so using the planner type as an input makes more sense to me.